### PR TITLE
Create human name trait

### PR DESCRIPTION
**Pull request by TW80000**
This looks pointless on its own, but it this got merged then in the nanopay repo we could use this for contacts as originally suggested here: https://github.com/nanoPayinc/NANOPAY/pull/3358/files/7ca73a988330a11afd0316e173e1a408fba3aa33#r213310409

### CODE REVIEW

File: src/foam/nanos/auth/HumanNameTrait.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    {
+      class: 'String',
+      name: 'middleName',
+      documentation: 'Middle name of user.',
+      validateObj: function(middleName) {
+        if ( middleName.length > 70 ) {
+          return 'Middle name cannot exceed 70 characters.';
+        }
+
+        if ( /\d/.test(middleName) ) {
```
Comment (kgrgreer) [inline src/foam/nanos/auth/HumanNameTrait.js:36]: Reminds me: we should add a 'pattern' property to Property.

Comment (adamvy) [reply src/foam/nanos/auth/HumanNameTrait.js:36]: Agree, although I disagree with the decision to disallow numbers in people's names.  Who are we to say that's valid or invalid.  I'm sure somebody somewhere has a digit in their name and I really don't see the value this extra "validation" is providing us.

File: src/foam/nanos/auth/User.js [hunk 2]
```diff
@@ -46,64 +47,10 @@ foam.CLASS({
...
-      expression: function(firstName, middleName, lastName) {
-        return middleName != '' ? firstName + ' ' + middleName + ' ' + lastName : firstName + ' ' + lastName;
-      },
-      javaGetter: `
-        return ! getMiddleName().equals("")
-          ? getFirstName() + " " + getMiddleName() + " " + getLastName()
-          : getFirstName() + " " + getLastName();
-      `,
-    },
+    'firstName',
```
Comment (adamvy) [inline src/foam/nanos/auth/User.js:59]: Why are we defining the properties again?

Comment (adamvy) [reply src/foam/nanos/auth/User.js:59]: Oh presumably for the java generation issue that Kirk was talking about.  Can we file an issue about that and then reference that issue from a //TODO comment here?

Comment (TW80000) [reply src/foam/nanos/auth/User.js:59]: Yes, it's because of the java generation problem. I'll do that now.

Comment (TW80000) [reply src/foam/nanos/auth/User.js:59]: Issue: https://github.com/foam-framework/foam2/issues/1529

Comment (TW80000) [reply src/foam/nanos/auth/User.js:59]: TODO comment added in 6c69df9

**Pull request merged**